src/ClusterNodesParser.h

```diff
@@ -55,6 +57,33 @@ class ClusterNodesParser
         end = mSlotEnd;
         return begin >= 0 && begin < end;
     }
+    bool validAddr() const {
+        if (mAddr.empty()) {
+            return false;
+        }
+
+        const char* host = "";
+        const char* port = strrchr(mAddr, ':');
+        if (port) {
+            std::string tmp;
+            tmp.append(mAddr, port - mAddr);
+            host = tmp.c_str();
```

Comment:
the host will point an invalid address

Reply:
Now i have changed it to use mAddr data instead of pointer.